java/core/src/test/org/apache/orc/impl/TestBrotli.java

```diff
@@ -136,10 +136,12 @@ public void testDirectDecompress() {
       // write bytes to heap buffer.
       assertTrue(brotliCodec.compress(in, out, null,
           brotliCodec.getDefaultOptions()));
+      int position = out.position();
       out.flip();
       // copy heap buffer to direct buffer.
       directOut.put(out.array());
       directOut.flip();
+      directOut.limit(position);
```

Comment:
May be I didn't understand the issue correctly, but from [documentation](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/ByteBuffer.html#flip()) of flip(), limit is set by it.

"Flips this buffer. The limit is set to the current position and then the position is set to zero"

Reply:
`out` is compressed by `brotliCodec`, and the length after compression is 217, but the `out.array()` used here is still the 10000 length of out, so the length that can be read after `directOut.flip()` is 10000, not 217

Reply:
Makes sense. Thanks !